interface/ceed-operator.c

```diff
@@ -303,7 +303,28 @@ int CeedOperatorAssembleLinearQFunction(CeedOperator op, CeedVector *assembled,
       return CeedError(ceed, 1, "At least one non-collocated basis required");
     // LCOV_EXCL_STOP
   }
-  ierr = op->AssembleLinearQFunction(op, assembled, rstr, request);
+  if (op->AssembleLinearQFunction) {
+    ierr = op->AssembleLinearQFunction(op, assembled, rstr, request);
+  } else {
+    // Reference Ceed
+    Ceed ceedref;
+    ierr = CeedInit("/cpu/self/ref/serial", &ceedref);
+
+    // Reference Operator
+    CeedOperator opref;
+    ierr = CeedCalloc(1, &opref); CeedChk(ierr);
+    memcpy(&opref, &op, sizeof(op));
+    opref->data = NULL;
+    ierr = ceedref->OperatorCreate(opref); CeedChk(ierr);
+
+    // Assemble
+    ierr = op->AssembleLinearQFunction(opref, assembled, rstr, request);
+
+    // Cleanup
+    ierr =  opref->Destroy(opref); CeedChk(ierr);
```

Comment:
Does the copy above mean that some arrays could be freed by `opref->Destroy`? Perhaps it's okay with strict programming practices.

Reply:
In theory, `opref->Destroy` should only destroy backend specific data, so I think we are ok there.

Where we are running into issues is with data synchronization inside the assembly of the QF. i.e. I will update a work vector's contents, but the vector that the work vector looks into does not know that its data has been updated.

Reply:
Perhaps some sort of Vec+offset workflow would help with this. I would drop the work vectors and just keep track of an index range to update for each input field/dim.